server/plugin.go

```diff
@@ -68,14 +75,33 @@ func (p *Plugin) OnActivate() error {
 		return err
 	}
 
-	bot, apperr := p.API.CreateBot(&model.Bot{
-		Username:    "cloud",
+	bot, appErr := p.API.CreateBot(&model.Bot{
+		Username:    botName,
 		DisplayName: "Cloud",
 		Description: "Created by the Mattermost Private Cloud plugin.",
 	})
-	if apperr != nil {
-		return errors.Wrap(apperr, "failed to ensure github bot")
+
+	if appErr != nil {
+		if !strings.Contains(appErr.Error(), "account with that username already exists") {
```

Comment:
Is there a better way to check this? I think that `AppError` contains a response status code, not sure how exactly MM handles that but perhaps it responds with something like 409?

Reply:
No, this API already changed behavior out from under us (it used to complete without error if the bot already existed) and I don't see a special error code for this case in the API documentation https://api.mattermost.com/#operation/CreateBot

As far as I can tell there's no cleaner way to differentiate between these errors.

Reply:
There is a helper method for this: https://github.com/mattermost/mattermost-plugin-playbooks/blob/4f9fb31bc1f1ae1672a35f9056b2a44164eda9ca/server/plugin.go#L67

We should probably switch to that. Could be a follow-up though if we wanted.

Reply:
That helper method doesn't work the way you think it does, anymore, @gabrieljackson . That's the reason for this change in the first place. That method returns an error when the bot already exists